src/NHibernate/Linq/Visitors/ResultOperatorProcessors/ProcessFetch.cs

```diff
@@ -61,20 +82,21 @@ private void Process(FetchRequestBase resultOperator, QueryModelVisitor queryMod
 							memberPath,
 							tree.TreeBuilder.Ident(innerFetch.RelationMember.Name));
 
-						Process(innerFetch, queryModelVisitor, tree, memberPath, componentType.Subtypes[subTypeIndex]);
+						Process(innerFetch, queryModelVisitor, tree, memberPath, currentNode, componentType.Subtypes[subTypeIndex]);
 					}
 
 					return;
 				}
 			}
 
 			var alias = queryModelVisitor.Model.GetNewName("_");
-			tree.AddFromClause(tree.TreeBuilder.LeftFetchJoin(memberPath, tree.TreeBuilder.Alias(alias)));
+			currentNode = tree.TreeBuilder.LeftFetchJoin(memberPath, tree.TreeBuilder.Alias(alias));
```

Comment:
For the `FetchManyRequest` the `currentNode` is always set, so the null check above does not matter.

Reply:
It seems not only `FetchManyRequest` but all fetch requests except components.

That got me thinking that `tree.GetFromRangeClause()` call should be moved inside component block as it's the only case it's needed.

Reply:
It is required only for property fetching so that the node is added to the correct from/join node. The applied change is valid, so I am marking this as resolved.